client/src/LayoutComponents/Home/Home.js

```diff
@@ -1,40 +1,84 @@
+  state = {
+    city: null,
+  };
 
-const Home = () => {
+  render() {
+    const { city } = this.state;
+    const { history } = this.props;
+    console.log(city, "please work!");
     return (
-        <div>
-            <section className="home-container">
-                <div className="home-intro">
-                    <p className="page-title"><b>Gaza Sky Geeks</b><br /> Code Academy Wiki</p>
-                    <p className="description">The GSG Code Academy is Palestine’s first immersive, 6-month career accelerating web development bootcamp.</p>
-                    <a href="https://gazaskygeeks.com/code/" target="_blank" rel="noopener noreferrer"><button className="join-btn">Join Us</button></a>
-                </div>
-                <img src={computer} alt="" />
-            </section>
+      <div>
+        <section className="home-container">
+          <div className="home-intro">
+            <p className="page-title">
+              <b>Gaza Sky Geeks</b>
+              <br /> Code Academy Wiki
+            </p>
+            <p className="description">
+              The GSG Code Academy is Palestine’s first immersive, 6-month
+              career accelerating web development bootcamp.
+            </p>
+            <a
+              href="https://gazaskygeeks.com/code/"
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+              <button className="join-btn">Join Us</button>
+            </a>
+          </div>
+          <img src={computer} alt="" />
+        </section>
 
-            <section className="cohorts-container">
-                <span className='cohorts-title'>Code Academy Cohorts</span>
-                <div className="cohorts">
-                    <img src={girl} alt="" />
-                    <div className="cohorts-btns">
-                        <button className="cohort-btn">Gaza</button>
-                        <button className="cohort-btn">Khalil</button>
-                    </div>
-                    <img src={boy} alt="" />
-                </div>
-            </section>
+        <section className="cohorts-container">
+          <span className="cohorts-title">Code Academy Cohorts</span>
+          <div className="cohorts">
+            <img src={girl} alt="" />
+            <div className="cohorts-btns">
+              <Link>
+                <button
+                  className="cohort-btn"
+                  onClick={() => {
+                    this.setState({ city: "gaza" });
+                  }}
+                >
+                  Gaza
+                </button>
+              </Link>
+              <Link>
+                <button
+                  className="cohort-btn"
+                  onClick={() => {
+                    this.setState({ city: "Kalel" });
+                  }}
+                >
+                  Khalil
+                </button>
+              </Link>
+              {city === "gaza" || city === "Kalel"
```

Comment:
@bayan-404 Instead of using click handlers to change state, and then doing history.push here in render, it would be simpler to have click handler functions which do the history.push.
First, inside this Home component, add new click handler functions:
```
gotoKhalil = () => {
  this.props.history.push('/cohorts/khalil');
}

gotoGaza = () => {
  this.props.history.push('/cohorts/gaza');
}
```
Then change your two button click handlers to call these functions:
```
<button className="cohort-btn" onClick={this.gotoGaza}>
  Gaza
</button>
<button className="cohort-btn" onClick={this.gotoKhalil}>
  Khalil
</button>
```
And then you can delete city from this.state.

Reply:
thank you for reviewing @tomduggan85  I appreciate it  
the reason why I use state here is to get the cohorts data for each city from the data base 

![datagaza](https://user-images.githubusercontent.com/57558867/81008303-6c65cb00-8e5b-11ea-91de-e1459db08de1.png)

Reply:
Ah okay sorry, I didn't see that.  I think you could keep that state then, but still do the click handling refactor I mentioned, to move the history.push outside of render().  But it's not an incredibly urgent refactor to make, so maybe it's an idea for next time :)

Reply:
I'll consider that ,thank you